kratos/spatial_containers/geometrical_objects_bins.h

```diff
@@ -343,6 +362,20 @@ class GeometricalObjectsBins
 
     }
 
+    template<typename TCellSizesType>
+    void AssignCellSize(const TCellSizesType& rCellSize) {
+        std::array<double, 3> lengths;
+        for (int i = 0; i < Dimension; i++) {
+            lengths[i] = mBoundingBox.GetMaxPoint()[i] - mBoundingBox.GetMinPoint()[i];
+        }
+
+        for (int i = 0; i < Dimension; i++) {
+            mNumberOfCells[i] = static_cast<std::size_t>(lengths[i] / rCellSize[i]);
```

Comment:
Consider floating point errors when casting. I'd round to the nearest integer instead (+ add a check in debug mode that verifies whether the value really is close to an integer).

Reply:
I added a round but i still dont check that it is close to an integer cause I dont know how to check this fact

Reply:
I don't think the check is needed here. But I would check if the NumberOfCells is zero and I would assign 1 to it:

```c++
std::size_t number_of_cells = static_cast<std::size_t>(std::round(lengths[i] / rCellSize[i]));
mNumberOfCells[i] = (number_of_cells >= 1) ? number_of_cells : 1;
```